src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -0,0 +1,547 @@
+    };
+
+    const reduced: AccType = spanChildren.reduce(
+      (acc: AccType, spanChild) => {
+        const key = `${traceID}${spanChild.span_id}`;
+
+        const results = this.renderSpan({
+          treeDepth: treeDepth + 1,
+          numOfHiddenSpansAbove: acc.numOfHiddenSpansAbove,
+          span: spanChild,
+          spanID: spanChild.span_id,
+          traceID,
+          lookup,
+          generateBounds,
+          pickSpanBarColour,
+        });
+
+        acc.renderedSpanChildren.push(
+          <React.Fragment key={key}>{results.spanTree}</React.Fragment>
+        );
+
+        acc.numOfHiddenSpansAbove = results.numOfHiddenSpansAbove;
+
+        return acc;
+      },
+      {
+        renderedSpanChildren: [],
+        numOfHiddenSpansAbove: isCurrentSpanHidden ? numOfHiddenSpansAbove + 1 : 0,
+      }
+    );
+
+    const showHiddenSpansMessage = !isCurrentSpanHidden && numOfHiddenSpansAbove > 0;
+
+    const hiddenSpansMessage = showHiddenSpansMessage ? (
+      <SpanRowMessage>
+        <span>Number of hidden spans: {numOfHiddenSpansAbove}</span>
+      </SpanRowMessage>
+    ) : null;
+
+    return {
+      numOfHiddenSpansAbove: reduced.numOfHiddenSpansAbove,
+      spanTree: (
+        <React.Fragment>
+          {hiddenSpansMessage}
+          <Span
+            span={span}
+            generateBounds={generateBounds}
+            treeDepth={treeDepth}
+            numOfSpanChildren={spanChildren.length}
+            renderedSpanChildren={reduced.renderedSpanChildren}
+            spanBarColour={spanBarColour}
+          />
+        </React.Fragment>
+      ),
+    };
+  };
+
+  renderRootSpan = (): RenderedSpanTree => {
+    const {dragProps, trace} = this.props;
+
+    // TODO: ideally this should be provided
+    const rootSpan: SpanType = {
+      trace_id: trace.traceID,
+      parent_span_id: void 0,
+      span_id: trace.rootSpanID,
+      start_timestamp: trace.traceStartTimestamp,
+      timestamp: trace.traceEndTimestamp,
+      same_process_as_parent: true,
+      op: 'transaction',
+      data: {},
+    };
+
+    const COLORS = ['#e9e7f7', '#fcefde', '#fffbee', '#f1f5fb'];
+    let current_index = 0;
+
+    const pickSpanBarColour = () => {
+      const next_colour = COLORS[current_index];
+
+      current_index++;
+      current_index = current_index % COLORS.length;
+
+      return next_colour;
+    };
+
+    const generateBounds = boundsGenerator({
+      traceStartTimestamp: trace.traceStartTimestamp,
+      traceEndTimestamp: trace.traceEndTimestamp,
+      viewStart: dragProps.viewWindowStart,
+      viewEnd: dragProps.viewWindowEnd,
+    });
+
+    return this.renderSpan({
+      treeDepth: 0,
+      numOfHiddenSpansAbove: 0,
+      span: rootSpan,
+      spanID: rootSpan.span_id,
+      traceID: rootSpan.trace_id,
+      lookup: trace.lookup,
+      generateBounds,
+      pickSpanBarColour,
+    });
+  };
+
+  render() {
+    const {spanTree, numOfHiddenSpansAbove} = this.renderRootSpan();
```

Comment:
Generally, our `render*` methods return a React element, can we rename the method to something else